testsuite/forge-cli/src/main.rs

```diff
@@ -1864,13 +1866,15 @@ fn realistic_network_tuned_for_throughput_test() -> ForgeConfig {
             }
         }))
         .with_genesis_helm_config_fn(Arc::new(move |helm_values| {
+            let mut on_chain_execution_config = OnChainExecutionConfig::default_for_genesis();
+            if let OnChainExecutionConfig::V4(config_v4) = &mut on_chain_execution_config {
```

Comment:
should we have a match in code , so that when adding a new type, compiling will not pass?